pkg/ocibundle/sif/bundle_linux.go

```diff
@@ -179,11 +180,23 @@ func FromSif(image, bundle string, writable bool) (ocibundle.Bundle, error) {
 		writable: writable,
 	}
 	s.bundlePath, err = filepath.Abs(bundle)
+	if err == nil {
+		stat, statErr := os.Stat(s.bundlePath)
+		if statErr == nil {
+			if !stat.IsDir() {
+				statErr = errors.New("bundle path is not a directory")
+			}
+		}
+		err = statErr
```

Comment:
@cclerget this function is overloaded without documentation. We should refactor to make these higher level design assumptions clearer. This is not a breaking change AFAIK.    If it is a breaking change, then the unit tests and other tests must be updated to detect such issues

Reply:
It is a breaking change for a caller so far, please look at https://github.com/apptainer/apptainer/blob/3f4d2c988d80d70f1f81c0760ec331a07bcb0b30/internal/app/apptainer/oci_mount_linux.go#L18
by not ignoring the `not exist` error, it will break `apptainer oci mount` which was working without requiring a bundle directory to exist, with `err = statErr`, `apptainer oci mount` will always fail if bundle doesn't exist.
So two solutions to fix this, create the directory (and parent) in `oci_mount_linux.go` prior to the call or reset `statErr` to nil when `os.IsNotExist(statErr)` return true in `FromSIf`.

This is the change requested initially

Reply:
removed all changes to `pkg/ocibundle/sif/*`